### PR TITLE
add libspng

### PR DESCRIPTION
**Pull request by totaam**
This library can be used by [`xpra`](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-python-xpra/PKGBUILD) to speedup `png` (de)compression.

### CODE REVIEW

File: mingw-w64-libspng/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+license=('spdx:BSD-2-Clause')
+depends=("${MINGW_PACKAGE_PREFIX}-zlib")
+makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
+             "${MINGW_PACKAGE_PREFIX}-cmake"
+             "${MINGW_PACKAGE_PREFIX}-ninja")
+source=(${_realname}-${pkgver}.tar.gz::https://github.com/randy408/libspng/archive/refs/tags/v${pkgver}.tar.gz)
+sha256sums=('47ec02be6c0a6323044600a9221b049f63e1953faf816903e7383d4dc4234487')
+# validpgpkeys=('6B0E6B70976DE303EDF2F601F9C3D6BDB8232B5D')
+
+prepare() {
```
Comment (ognevny) [inline mingw-w64-libspng/PKGBUILD:21]: `prepare()` function is useless here, it can be removed

File: mingw-w64-libspng/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+build() {
+  cd "${srcdir}/${_realname}-${pkgver}"
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+    "${MINGW_PREFIX}"/bin/cmake.exe \
+      -GNinja \
+      -DCMAKE_INSTALL_PREFIX="${MINGW_PREFIX}" \
+      "${extra_config[@]}" \
+      ../${_realname}-${pkgver}
```
Comment (ognevny) [inline mingw-w64-libspng/PKGBUILD:32]: please, follow the [template](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-PKGBUILD-templates%2FPKGBUILD.CMake-tarball.sh) so `extra_config` will be set

File: mingw-w64-libspng/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Maintainer: Antoine Martin <totaam@xpra.org>
+
+_realname=libspng
+pkgbase=mingw-w64-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+pkgver=0.7.4
+pkgrel=1
+pkgdesc="a C library for reading and writing Portable Network Graphics (PNG) format files with a focus on security and ease of use (mingw-w64)"
+arch=('any')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
```
Comment (ognevny) [inline mingw-w64-libspng/PKGBUILD:10]:
```suggestion
mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
```

File: mingw-w64-libspng/PKGBUILD (hunk 1 above)
Comment (totaam) [reply mingw-w64-libspng/PKGBUILD]: Done: 1d680806afd1e958101e6a3819bc965482004b4e

File: mingw-w64-libspng/PKGBUILD (hunk 2 above)
Comment (totaam) [reply mingw-w64-libspng/PKGBUILD:27]: @ognevny sorry, I've tried and cmake is not happy with it - I'm out of my depth here:
```
-- Build files have been written to: E:/MINGW-packages/mingw-w64-libspng/src/build-MINGW64
[1/6] Building C object CMakeFiles/example.dir/examples/example.c.obj
[2/6] Building C object CMakeFiles/spng_static.dir/spng/spng.c.obj
[3/6] Building C object CMakeFiles/spng.dir/spng/spng.c.obj
[4/6] Linking C static library libspng_static.a
[5/6] Linking C shared library libspng.dll
[6/6] Linking C executable example.exe
==> Starting package()...
CMake Error: Not a file: E:/MINGW-packages/mingw-w64-libspng/src/libspng-0.7.4/build-MINGW64/cmake_install.cmake
CMake Error: Error processing file: E:/MINGW-packages/mingw-w64-libspng/src/libspng-0.7.4/build-MINGW64/cmake_install.cmake
==> ERROR: A failure occurred in package().
    Aborting...
```

Comment (Biswa96) [reply mingw-w64-libspng/PKGBUILD]: I have added necessary stuff to follow pkgbuild template.

**Pull request merged**